src/renderer/src/hooks/stores/mapeoDeviceStore.ts

```diff
@@ -0,0 +1,122 @@
+import { create } from 'zustand'
+
+type DeviceType = 'mobile' | 'desktop'
+
+export type Role = 'coordinator' | 'participant' | 'waitingInvite'
+
+type MapeoDevice =
+  | {
+      name: string
+      deviceId: string
+      deviceType: DeviceType
+      pastDeviceNames?: string[]
+      role: Extract<Role, 'waitingInvite'>
+    }
+  | {
+      name: string
+      deviceId: string
+      deviceType: DeviceType
+      pastDeviceNames?: string[]
+      role: Extract<Role, 'coordinator' | 'participant'>
+      dateAdded: Date
+    }
+
+type MapeoDeviceState = {
+  devices: Record<string, MapeoDevice>
+  actions: {
+    addDeviceToProject: (deviceId: string, role: Extract<Role, 'coordinator' | 'participant'>) => void
+    removeDeviceFromProject: (deviceId: string) => void
+    changeDeviceRole: (deviceId: string, newRole: Extract<Role, 'coordinator' | 'participant'>) => void
+  }
+}
+
+export const mapeoDeviceStore = create<MapeoDeviceState>()((set) => ({
```

Comment:
Since `create` creates a hook, would be helpful to use conventional naming

```suggestion
export const useMapeoDevicesStore = create<MapeoDeviceState>()((set) => ({
```